consensus/consensus_v2.go

```diff
@@ -761,6 +768,29 @@ func (consensus *Consensus) finalizeCommits() {
 func (consensus *Consensus) onCommitted(msg *msg_pb.Message) {
 	consensus.getLogger().Debug().Msg("[OnCommitted] Receive committed message")
 
+	// TODO: this is temp hack for update new node's committee information; remove it after staking and resharding finished
+	if consensus.mode.Mode() == Listening {
+		recvMsg, err := ParsePbftMessage(msg)
+		if err != nil {
+			consensus.getLogger().Warn().Msg("[OnCommitted] unable to parse msg")
+			return
+		}
+		if recvMsg.BlockNum%core.BlocksPerEpoch == core.BlocksPerEpoch-1 {
```

Comment:
better add a comment here on why doing this check

Reply:
why don't you use the core.IsEpochLastBlock function? Or create a similar function to make it more readable.

Reply:
Because IsEpochLastBlock needs a block as input. here we don't have the block

Reply:
comment added